AdobeStockImageAdminUi/view/adminhtml/web/js/action/authorization.js

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+define([], function () {
+    'use strict';
+    /**
+     * Consts of response
+     *
+     * RESPONSE_PATTERN - pattern of response
+     * RESPONSE_CODE_INDEX index of response code
+     * RESPONSE_MESSAGE_INDEX index of response message
+     * RESPONSE_SUCCESS_CODE success code
+     */
+    var RESPONSE_PATTERN = /auth\[code=(success|error);message=(.+)\]/,
+        RESPONSE_CODE_INDEX = 1,
+        RESPONSE_MESSAGE_INDEX = 2,
+        RESPONSE_SUCCESS_CODE = 'success';
+
+    /**
+     * Build window params
+     *
+     * @param {Object} windowParams
+     * @returns {String}
+     */
+    function buildWindowParams (windowParams) {
+        var output = '',
+            coma = '',
+            paramName,
+            paramValue;
+
+        for (paramName in windowParams) {
+            paramValue = windowParams[paramName];
+            output += coma + paramName + '=' + paramValue;
+            coma = ',';
+        }
+
+        return output;
+    };
+
+    /**
+     * Authorization process
+     *
+     * @param {Object} authConfig
+     * @return {Promise}
+     */
+    function authorizationProcess (authConfig) {
```

Comment:
Let's use this as an anonymous function to be compatible with existing Magento actions.

I.e. `return function (authConfig) {...}`